src/chunk.c

```diff
@@ -4927,7 +4940,40 @@ ts_osm_chunk_range_is_invalid(int64 range_start, int64 range_end)
 int32
 ts_chunk_get_osm_slice_id(int32 chunk_id, int32 time_dim_id)
 {
-	Chunk *chunk = ts_chunk_get_by_id(chunk_id, true);
+	ScanKeyData scankey[1];
+	static const DisplayKeyData displaykey[1] = {
+		[0] = { .name = "id", .as_string = DatumGetInt32AsString },
+	};
+
+	/*
+	 * Perform an index scan on chunk id.
+	 */
+	ScanKeyInit(&scankey[0],
+				Anum_chunk_idx_id,
+				BTEqualStrategyNumber,
+				F_INT4EQ,
+				Int32GetDatum(chunk_id));
+
+	/*
```

Comment:
I am wondering if it would be better to simply allow the caller to pass in the locklevel in order to get the "right" lock from the start rather than not locking and then upgrading.

I guess the lock upgrade from FOR KEY SHARE was the problem to begin with so if we allow explicitly grabbing either FOR KEY SHARE _or_ FOR UPDATE from the start when needed that should solve the problem.